### PR TITLE
Fix CI Python always at 3.10.

### PR DESCRIPTION
**Pull request by hameerabbasi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
           activate-environment: sparse-dev
           allow-softlinks: true
           environment-file: ci/environment.yml
-          python-version: '3.10'
+          python-version: ${{ matrix.python }}
```
Comment (mtsokol) [inline .github/workflows/ci.yml:81]: Does it mean that `docs` stage will run against three different python versions? Is it beneficial? (I think that e.g. NumPy runs docs only against one python version)

Comment (mtsokol) [reply .github/workflows/ci.yml:81]: Hmm but `docs` job doesn't define `matrix` entry.

Comment (mtsokol) [reply .github/workflows/ci.yml:81]: Where is the `matrix.python` coming from here?

**Comment by hameerabbasi** [comment]
My bad, I thought this was during the `tests` job.

**Pull request closed without merging**